internal/manager/setup.go

```diff
@@ -309,20 +316,22 @@ func (c *Config) adminAPIClientFromServiceDiscovery(ctx context.Context, httpcli
 	err = retry.Do(func() error {
 		s, err := adminapi.GetAdminAPIsForService(ctx, kubeClient, kongAdminSvcNN, sets.New(c.KondAdminSvcPortNames...))
 		if err != nil {
-			return err
+			return retry.Unrecoverable(err)
 		}
 		if s.Len() == 0 {
-			return fmt.Errorf("no endpoints for kong admin service: %q", kongAdminSvcNN)
+			return NoAvailableEndpointsError{serviceNN: kongAdminSvcNN}
 		}
 		adminAPIs = s.UnsortedList()
 		return nil
 	},
 		retry.Context(ctx),
-		retry.Attempts(60),
+		retry.Attempts(0),
 		retry.DelayType(retry.FixedDelay),
 		retry.Delay(time.Second),
 		retry.OnRetry(func(_ uint, err error) {
-			logrus.New().WithError(err).Error("failed to create kong client(s)")
+			if !errors.As(err, &NoAvailableEndpointsError{}) {
+				logger.Error(err, "failed to create kong client(s)")
```

Comment:
```suggestion
				logger.Error(err, fmt.Sprintf("No endpoints for %q service discovered, retrying", kongAdminSvcNN))
```

Reply:
Just to note: with the code above this retry block is not really needed right? We have 3 options: 

- the error is `nil`: no-op
- the error is `NoAvailableEndpointsError`: we retry but we don't log because we don't want to spam logs 
- the error is something else: the only error as of now which we can get is `retry.Unrecoverable` error which will prevent retries from happening

I'm OK with leaving this block as is though. Might come in handy in the future.

Reply:
I think there could be possibly other errors in the future, so I wanted to keep it as-is.